src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -45,19 +48,44 @@ public int GetJobTypeConcurrencyLimit(string jobTypeName)
             ?.ConcurrencyPolicy
             ?.MaxDegreeOfParallelism ?? 1;
 
-    public void RemoveByName(string jobName) => Remove(allJobs.FirstOrDefault(j => j.CustomName == jobName));
+    public string? RemoveByName(string jobName)
+    {
+        EnsureCanBeRemoved(j => j.CustomName == jobName);
+
+        var jobDefinitionFullName = FindJobDefinition(jobName)?.JobFullName;
+
+        if (jobDefinitionFullName is null)
+        {
+            return null;
+        }
+
+        Remove(allJobs.FirstOrDefault(j => j.CustomName == jobName));
+
+        return jobDefinitionFullName;
+    }
 
-    public void RemoveByType(Type type)
+    public string? RemoveByType(Type type)
     {
+        EnsureCanBeRemoved(j => j.Type == type);
+
+        var jobDefinitionFullName = FindFirstJobDefinition(type)?.JobFullName;
+
+        if (jobDefinitionFullName is null)
+        {
+            return null;
+        }
+
         var jobDefinitions = FindAllJobDefinition(type);
 
         foreach (var jobDefinition in jobDefinitions)
         {
             Remove(jobDefinition);
         }
+
+        return jobDefinitionFullName;
     }
 
-    public JobDefinition AddDynamicJob(
+    public static JobDefinition CreateDynamicJob(
```

Comment:
That is information which is solely relying on `JobDefinition` - wouldn't it make more sense to move it there?

Reply:
Fixed. It was actually simpler to just inline it.